### PR TITLE
feat: Define import API and metadata

### PR DESCRIPTION
**Pull request by bigsheeper**
Define the new rpc and metadata for ImportV2.

see also: https://github.com/milvus-io/milvus/issues/28521

### CODE REVIEW

File: internal/proto/data_coord.proto [hunk 1]
```diff
@@ -737,3 +743,71 @@ message ChannelOperationProgressResponse {
   ChannelWatchState state = 3;
   int32 progress = 4;
 }
+
+enum ImportState {
+  Pending = 0;
```
Comment (czs007) [inline internal/proto/data_coord.proto:748]: None=0;
Pending = 1;

File: internal/proto/data_coord.proto [hunk 2]
```diff
@@ -737,3 +743,71 @@ message ChannelOperationProgressResponse {
...
+  repeated ImportFileInfo file_infos = 6;
+  repeated SegmentInfo segment_infos = 7;
+}
+
+message DropImportRequest {
+  int64 requestID = 1;
+  int64 taskID = 2;
+}
+
+message ImportFileInfo {
```
Comment (czs007) [inline internal/proto/data_coord.proto:795]: split ImportFieldInfo, do not merge PreImport and Import

**Comment by bigsheeper** [comment]
rerun ut

File: internal/datacoord/mock_test.go [hunk 3]
```diff
@@ -295,6 +295,26 @@ func (c *mockDataNodeClient) CheckChannelOperationProgress(ctx context.Context,
...
 
+func (c *mockDataNodeClient) PreImport(ctx context.Context, req *datapb.PreImportRequest, opts ...grpc.CallOption) (*commonpb.Status, error) {
+	return &commonpb.Status{ErrorCode: commonpb.ErrorCode_Success}, nil
+}
+
+func (c *mockDataNodeClient) ImportV2(ctx context.Context, req *datapb.ImportRequest, opts ...grpc.CallOption) (*commonpb.Status, error) {
+	return &commonpb.Status{ErrorCode: commonpb.ErrorCode_Success}, nil
+}
+
+func (c *mockDataNodeClient) GetPreImportState(ctx context.Context, req *datapb.GetPreImportStateRequest, opts ...grpc.CallOption) (*datapb.GetPreImportStateResponse, error) {
```
Comment (czs007) [inline internal/datacoord/mock_test.go:306]: DescribePreImport?

File: internal/datacoord/mock_test.go [hunk 4]
```diff
@@ -295,6 +295,26 @@ func (c *mockDataNodeClient) CheckChannelOperationProgress(ctx context.Context,
...
+	return &datapb.GetPreImportStateResponse{Status: merr.Success()}, nil
+}
+
+func (c *mockDataNodeClient) GetImportState(ctx context.Context, req *datapb.GetImportStateRequest, opts ...grpc.CallOption) (*datapb.GetImportStateResponse, error) {
```
Comment (czs007) [inline internal/datacoord/mock_test.go:310]: DescribeImport?

File: internal/proto/data_coord.proto [hunk 5]
```diff
@@ -737,3 +744,115 @@ message ChannelOperationProgressResponse {
...
+}
+
+message ImportRequest {
+  int64 requestID = 1;
+  int64 taskID = 2;
+  int64 collectionID = 3;
+  int64 partitionID = 4;
+  schema.CollectionSchema schema = 5;
+  repeated ImportFile import_files = 6;
+  repeated int64 segmentIDs = 7;
```
Comment (czs007) [inline internal/proto/data_coord.proto:783]: use Map<SegmentID> channelss?
Map<segmentID> IDRange?

File: internal/proto/data_coord.proto [hunk 6]
```diff
@@ -119,6 +119,13 @@ service DataNode {
   rpc FlushChannels(FlushChannelsRequest) returns(common.Status) {}
   rpc NotifyChannelOperation(ChannelOperationsRequest) returns(common.Status) {}
   rpc CheckChannelOperationProgress(ChannelWatchInfo) returns(ChannelOperationProgressResponse) {}
+
+  // import v2
+  rpc PreImport(PreImportRequest) returns(common.Status) {}
+  rpc ImportV2(ImportRequest) returns(common.Status) {}
+  rpc GetImportState(GetImportStateRequest) returns(GetImportStateResponse) {}
```
Comment (czs007) [inline internal/proto/data_coord.proto:126]: QueryImport or QueryImportJob?

File: internal/proto/data_coord.proto [hunk 7]
```diff
@@ -119,6 +119,13 @@ service DataNode {
...
+  rpc GetPreImportState(GetPreImportStateRequest) returns(GetPreImportStateResponse) {}
```
Comment (czs007) [inline internal/proto/data_coord.proto:127]: QueryPreImport or QueryPreImportJob?

**Pull request merged**